### DISCUSSION TITLE
Write value with MQTT into ebusd

### DISCUSSION THREAD

**Discussion by wseifert** [john30/ebusd#1320] [Q&A]
I want to write a value (room temperature) via mosquitto MQTT from openhab to ebusd and further to my Ochner OTE.

I am successfull using ebusctl:

`ebusctl write -c 22420 room/temperature/set 24.0`

I can see immediately in MQTT explorer the result

- ebusd
  - global
    - 22420
      - room
        - temperature = 1;0;0d;°C;50.0;0.0;24.0
          - set = 24.0

and after a short time also on the OTE display.

But I am not successfull writing the value from openhab via mosquitto MQTT.

Here my openhab mqtt.things:

`Bridge mqtt:broker:mosquitto "Mosquitto MQTT Broker" @ "MQTT" [ 
    host="localhost", 
    secure=false, 
    port=1883
]
{
    Thing topic ebusd
    {
        Type number : temperature_room [ stateTopic="ebusd/22420/room/temperature", commandTopic="ebusd/22420/room/temperature/set", transformationPattern="JS:|(input).split(';')[6]"]
     }
 }`

I can see the new value in MQTT explorer write topic set (changed from 24.0 to 24.5)

`- ebusd
  - global
    - 22420
      - room
        - temperature = 1;0;0d;°C;50.0;0.0;24.0
          - set = 24.5`

but not on the read topic temperature and OTE display.

Maybe a wrong value type used by openhab?

I do not have any idea how to solve this problem, any help is welcome!

Thanks in advance
Werner

**Comment by wseifert** [comment]
I did more testing using MQTT Explorer:

publish raw value of 30.0 to ebusd/22420/room/temperature/set, result in ebusd.log:

`2024-08-07 15:35:05.525 [mqtt error] received unmatchable topic ebusd/22420/room/temperature/set`

to compare result publish same value to same topic with ebusctl (ebusctl write -c 22420 room/temperature/set 30.0) result in ebusdlog:

`2024-08-07 15:35:29.447 [update notice] sent write 22420 room/temperature/set QQ=31: 30.0`

here the definition in 15.22420.HW0102.SW0502.csv:

`r1,,room/temperature,00-001 Aktuelle Raumtemperatur,,,,0081004a,,,param,,,,,,tempt,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,`
`w,,room/temperature/set,00-001 Aktuelle Raumtemperatur,,,,0081004a,,,tempt,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,`

I attach my *.csv files.

Can anybody explain what is wrong publishing with MQTT Explorer?

Werner

**Comment by wseifert** [comment]
[_templates.csv](https://github.com/user-attachments/files/16530771/_templates.csv)
[15.22420.HW0102.SW0502.csv](https://github.com/user-attachments/files/16530772/15.22420.HW0102.SW0502.csv)
[broadcast.csv](https://github.com/user-attachments/files/16530773/broadcast.csv)

**Comment by john30** [comment]
you shouldn't use slashes in message and/or circuit names as this will break things at various locations.
furthermore, the /set suffix is handled by the MQTT handler of ebusd internally, so especially this is not supposed to be revealed in the message definition at all.
in general ebusd associates read and write messages with the same circuit and message name with each other, so that it will be readable and writable where the write desire in mqtt is signaled to ebusd using the extra /set suffix as described here:
https://github.com/john30/ebusd/wiki/3.3.-MQTT-client#mqtt-topics
